### PR TITLE
Extend ChatOps pack for other platforms

### PR DESCRIPTION
**Pull request by tonybaloney**
Some chat platforms either have no hubot support or poor support. The lowest common denominator is webhooks.
In order to setup a webhook, st2 needs a way to call the action alias matching and execute 

* `match_and_execute` - Returns an execution ID for a chatops text message
* `match` - Matches a string to an action alias
* `run` - Calls `match_and_execute` and passes execution ID to `post_result`

### CODE REVIEW

**Comment by tonybaloney** [comment]
this is done. please review

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/community/p1495394698726751

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/community/p1495399563090690

**Comment by Kami** [comment]
Nice work :+1: 

Will have a look shortly.

File: contrib/chatops/actions/match_and_execute.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        alias_match = ActionAliasMatch()
+        alias_match.command = text
+        alias, representation = self.client.managers['ActionAlias'].match(
+            alias_match)
+
+        execution = ActionAliasExecution()
+        execution.name = alias.name
+        execution.format = representation
+        execution.command = text
+        execution.source_channel = source_channel  # ?
```
Comment (Kami) [inline contrib/chatops/actions/match_and_execute.py]: Could default it to "runner_action" or similar.

It's meant to represent a source (CLI, Slack, etc.).

**Comment by Kami** [comment]
Nice work, the changes look good to me.

Also plan to try it out shortly :)

**Comment by emedvedev** [comment]
LGTM

**Comment by Kami** [comment]
Tested it all locally and confirmed it's working.

Nice work :+1: 

On a related note, would still be great to add some unit / integration tests for it in the future :)

**Pull request merged**